modules/margin-protocol/src/lib.rs

```diff
@@ -967,6 +1006,98 @@ impl<T: Trait> Module<T> {
 		Ok(())
 	}
 
+	/// get trader's max threshold based on opened positions after performing an action.
+	///
+	/// Return `RiskThreshold` or `Default` value.
+	fn _get_trader_max_threshold(who: &T::AccountId, action: Action<T>) -> RiskThreshold {
+		let new_position = match action.clone() {
+			Action::OpenPosition(p) => Some(p),
+			_ => None,
+		};
+		let mut trading_pairs: Vec<TradingPair> = <PositionsByTrader<T>>::iter(who)
+			.filter_map(|((_, position_id), _)| {
+				let position = Self::positions(position_id)?;
+				Some(position.pair)
+			})
+			.chain(new_position.map_or(vec![], |p| vec![p.pair]))
+			.collect();
+		trading_pairs.sort();
+		trading_pairs.dedup();
+
+		let trader_margin_call = trading_pairs
+			.iter()
+			.filter_map(|pair| Some(Self::trader_risk_threshold(pair).unwrap_or_default()))
+			.max_by_key(|threshold| threshold.margin_call)
+			.map(|x| x.margin_call)
+			.unwrap_or_default();
+		let trader_stop_out = trading_pairs
+			.iter()
+			.filter_map(|pair| Some(Self::trader_risk_threshold(pair).unwrap_or_default()))
+			.max_by_key(|threshold| threshold.stop_out)
+			.map(|x| x.stop_out)
+			.unwrap_or_default();
+
+		RiskThreshold {
+			margin_call: trader_margin_call,
+			stop_out: trader_stop_out,
+		}
+	}
+
+	/// get liquidity pool's max threshold based on opened positions after performing an action.
+	///
+	/// Return `RiskThreshold` or `Default` value.
+	fn _get_liquidity_pool_max_threshold(
+		pool_id: LiquidityPoolId,
+		action: Action<T>,
+	) -> (RiskThreshold, RiskThreshold) {
```

Comment:
```suggestion
	fn _enp_and_ell_risk_threshold_of_pool(
		pool_id: LiquidityPoolId,
	) -> (RiskThreshold, RiskThreshold) {
```

Same reason to trader.